homeassistant/config_entries.py

```diff
@@ -3266,26 +3266,22 @@ async def async_finish_flow(
     ) -> SubentryFlowResult:
         """Finish a subentry flow and add a new subentry to the configuration entry.
 
-        The flow.handler and the entry_id is the same thing to map flow with entry.
+        The flow.handler[0] and entry_id is the same thing to map flow with entry.
         """
         flow = cast(ConfigSubentryFlow, flow)
 
         if result["type"] != data_entry_flow.FlowResultType.CREATE_ENTRY:
             return result
 
-        entry = self.hass.config_entries.async_get_entry(flow.handler[0])
+        entry_id = flow.handler[0]
+        entry = self.hass.config_entries.async_get_entry(entry_id)
         if entry is None:
-            raise UnknownEntry(flow.handler[0])
+            raise UnknownEntry(entry_id)
 
         unique_id = result.get("unique_id")
         if unique_id is not None and not isinstance(unique_id, str):
             raise HomeAssistantError("unique_id must be a string")
 
-        if unique_id is not None and any(
-            subentry.unique_id == unique_id for subentry in entry.subentries.values()
-        ):
-            raise data_entry_flow.AbortFlow("already_configured")
```

Comment:
Why did we remove this check?

Reply:
Nevermind, I see there's a check in the call below.

Reply:
I moved it here: https://github.com/home-assistant/core/pull/117355/files#diff-bcb932c3549daf52e1d6d8b02211ed0cc15557f5d17968f33a9daacf7ddac81bR2393-R2401

Integrations should have a check in their subentry flow to not create duplicates, then we check it too, just before actually adding the subentry to the config entry.